pymove/tests/core/test_data_frame.py

```diff
@@ -0,0 +1,1102 @@
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+    ], columns=['lat', 'lonerr', 'datetime', 'id'], index=[0, 1, 2, 3]
+    )
+    try:
+        MoveDataFrame(data=df, latitude=LATITUDE, longitude=LONGITUDE, datetime=DATETIME)
+        raise AssertionError('AttributeError error not raised by MoveDataFrame')
+    except AttributeError:
+        pass
+
+    df = DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+    ], columns=['lat', 'lon', 'datetimerr', 'id'], index=[0, 1, 2, 3]
+    )
+    try:
+        MoveDataFrame(data=df, latitude=LATITUDE, longitude=LONGITUDE, datetime=DATETIME)
+        raise AssertionError('AttributeError error not raised by MoveDataFrame')
+    except AttributeError:
+        pass
+
+
+def test_lat():
+    move_df = _default_move_df()
+
+    lat = move_df.lat
+    srs = Series(
+        data=[39.984094, 39.984198, 39.984224, 39.984224],
+        index=[0, 1, 2, 3], dtype='float64', name='lat'
+    )
+    assert_series_equal(lat, srs)
+
+
+def test_lon():
+    move_df = _default_move_df()
+
+    lon = move_df.lon
+    srs = Series(
+        data=[116.319236, 116.319322, 116.319402, 116.319402],
+        index=[0, 1, 2, 3], dtype='float64', name='lon'
+    )
+    assert_series_equal(lon, srs)
+
+
+def test_datetime():
+    move_df = _default_move_df()
+
+    datetime = move_df.datetime
+    srs = Series(
+        data=['2008-10-23 05:53:05', '2008-10-23 05:53:06',
+              '2008-10-23 05:53:11', '2008-10-23 05:53:11'],
+        index=[0, 1, 2, 3], dtype='datetime64[ns]', name='datetime'
+    )
+    assert_series_equal(datetime, srs)
+
+
+def test_loc():
+    move_df = _default_move_df()
+
+    assert move_df.loc[0, TRAJ_ID] == 1
+
+    loc_ = move_df.loc[move_df[LONGITUDE] > 116.319321]
+    expected = DataFrame(data=[
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+    ], columns=['lat', 'lon', 'datetime', 'id'], index=[1, 2, 3]
+    )
+    assert_frame_equal(loc_, expected)
+
+
+def test_iloc():
+    move_df = _default_move_df()
+
+    expected = Series(
+        data=[39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1],
+        index=['lat', 'lon', 'datetime', 'id'], dtype='object', name=0
+    )
+
+    assert_series_equal(move_df.iloc[0], expected)
+
+
+def test_at():
+    move_df = _default_move_df()
+
+    assert move_df.at[0, TRAJ_ID] == 1
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574968)